### ISSUE TITLE
Add support for ON UPDATE <expr>

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#6132]
Some databases support an additional `ON UPDATE` clause in addition to the `DEFAULT` clause on data types, which evaluates a given expression and puts that as the value for the column.

MySQL calls this `ON UPDATE`, whereas Oracle calls this `GENERATED ALWAYS` (although not all expressions or data types are supported)

**Comment by lukaseder** [comment]
We already support computed columns, now, since jOOQ 3.16: https://www.jooq.org/doc/latest/manual/sql-building/column-expressions/computed-columns/

**Comment by lukaseder** [comment]
I closed this too eagerly. The `ON UPDATE` clause works subtly differently. It's equivalent to `GENERATED ALWAYS AS`,  but triggered only "`ON UPDATE`". To be investigated further.

**Comment by apatrida** [comment]
> It's equivalent to GENERATED ALWAYS AS ...

It is not actually.  `GENERATED ALWAYS AS` (or just `AS`) in MySQL only works on deterministic functions that return the same value for the same input.  You cannot use `CURRENT_TIMESTAMP` for example in these functions, but you can for `ON UPDATE`

from https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html

> Generated column expressions must adhere to the following rules. An error occurs if an expression contains disallowed constructs.
> 
> Literals, deterministic built-in functions, and operators are permitted. A function is deterministic if, given the same data in tables, multiple invocations produce the same result, independently of the connected user. Examples of functions that are nondeterministic and fail this definition: [CONNECTION_ID()](https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_connection-id), [CURRENT_USER()](https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_current-user), [NOW()](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_now).

Also of note, is that `ON UPDATE` I believe allows you to override the value on `UPDATE` where I am not sure `GENERATED ALWAYS AS` allows you to pass in a new value on `UPDATE` to override the generated value.

**Comment by apatrida** [comment]
Yes, checked. At least for MySQL you cannot use a column that is `GENERATED ALWAYS AS ... STORED` in an `UPDATE`, but you can for `ON UPDATE` column.

**Comment by lukaseder** [comment]
Yes, indeed. My definition of "equivalent" may have been overly lax. `ON UPDATE` is more of a very simplified trigger.
